bundle-hack/update_bundle.sh

```diff
@@ -1,5 +1,107 @@
+source ./container_digest.sh
+source ./bundle_vars.sh
+
+VERSION=$(cat VERSION)
+export VERSION
+
+# Check for environment variables pertaining to the bundle
+if  [ -z "${OPERATOR_IMAGE_PULLSPEC}" ] ||
+    [ -z "${OPERAND_IMAGE_PULLSPEC}" ] ||
+    [ -z "${KUBE_RBAC_PROXY_IMAGE_PULLSPEC}" ] ||
+    [ -z "${MANIFESTS_DIR}" ] ||
+    [ -z "${METADATA_DIR}" ] ||
+    [ -z "${SUPPORTED_OCP_VERSIONS}" ] ||
+    [ -z "${VERSION}" ]; then
+  echo "ERROR: Not all required environment variables are set"
+  echo "    OPERATOR_IMAGE_PULLSPEC"
+  echo "    OPERAND_IMAGE_PULLSPEC"
+  echo "    KUBE_RBAC_PROXY_IMAGE_PULLSPEC"
+  echo "    MANIFESTS_DIR"
+  echo "    METADATA_DIR"
+  echo "    SUPPORTED_OCP_VERSIONS"
+  echo "    VERSION"
+  exit 2
+fi
+
+# Corrected CSV_FILE variable definition with closing parenthesis
+CSV_FILE="${MANIFESTS_DIR:-/manifests}/external-dns-operator.clusterserviceversion.yaml"
+
+# Update direct references in CSV to match release targets
+sed -i -e "s|quay.io/openshift/origin-external-dns-operator:latest|${OPERATOR_PULLSPEC}|g" \
+       -e "s|k8s.gcr.io/external-dns/external-dns:.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/openshift/origin-external-dns:.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/openshift/origin-external-dns@.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/external-dns-operator/external-dns:.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/external-dns-operator/external-dns@.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/openshift/origin-kube-rbac-proxy:latest|registry.redhat.io/openshift4/ose-kube-rbac-proxy:v4.13|g" "${CSV_FILE}"
+
+export EPOC_TIMESTAMP=$(date +%s)
+export TARGET_CSV_FILE="${CSV_FILE}"
+
+# Combined all Python logic into a single block
+python3 - << CSV_UPDATE
+import os
+from sys import exit as sys_exit
+from datetime import datetime
+from ruamel.yaml import YAML
+
+yaml = YAML()
+
+def load_manifest(pathn):
+   if not pathn.endswith(".yaml"):
+      return None
+   try:
+      with open(pathn, "r") as f:
+         return yaml.load(f)
+   except FileNotFoundError:
+      print("File can not be found")
+      exit(6)
+
+def dump_manifest(pathn, manifest):
+   with open(pathn, "w") as f:
+      yaml.dump(manifest, f)
+   return
+
+timestamp = int(os.getenv('EPOC_TIMESTAMP'))
+datetime_time = datetime.fromtimestamp(timestamp)
+version = os.getenv('VERSION')
+replaces = os.getenv('REPLACES_VERSION')
+operator_pullspec = os.getenv('OPERATOR_IMAGE_PULLSPEC', '')
+csv = load_manifest(os.getenv('TARGET_CSV_FILE'))
+
+# Update CSV metadata and spec
+csv['metadata']['annotations']['createdAt'] = datetime_time.strftime('%Y-%m-%dT%H:%M:%S')
+csv['metadata']['annotations']['containerImage'] = operator_pullspec
+csv['metadata']['name'] = 'external-dns-operator.v{}'.format(version)
+csv['metadata']['annotations']['olm.skipRange'] = '<{}'.format(version)
+csv['spec']['version'] = version
+
+if replaces:
+    csv['spec']['replaces'] = 'external-dns-operator.v{}'.format(replaces)
+
```

Comment:
CVS's related images have to be filled in as it's not done automatically in Konflux ([example from ALBO](https://github.com/openshift/aws-load-balancer-operator/blob/911d90c16d5bc979570f522bc0bef5e561468858/bundle-hack/update_bundle.sh#L91-L103)).